File: 202_搜尋/TestGraphII/Stack.cs
using System;
namespace lesson202_01
{
    public class Stack
    {
        private int top;
        private int[] items;
        public Stack(int size)
        {
            top = -1;
            items = new int[size];
        }
        public bool IsEmpty()
        {
            if (top < 0)
                return true;
            return false;
        }
        public bool IsFull()
        {
            if (top >= items.Length - 1)
                return true;
            return false;
        }
		public void Push(int item)
		{
            if (!IsFull())
                items[++top] = item;
		}
        public int Pop()
        {
            if (IsEmpty())
                return -1;
            return items[top--];
        }
    }
}

File: 999_OA/ArithmeticByStack/Stack.java

class Stack{
	
	private final int capacity = 10;
	private int top;
	private char[] items;
	
	public Stack() {
		init(capacity);
	}
	public Stack(int capacity) {
		init(capacity);
	}
	private void init(int capacity) {
		top = -1;
		items = new char[capacity];
	}
	public boolean isEmpty() {
		return top < 0;
	}
	public boolean isFull() {
		return top >= items.length - 1;
	}
	public void push(char item) {
		if(isFull()) resize();
		items[++top] = item;
	}
	public char pop() {
		if(!isEmpty())
			return items[top--];
		return ' ';
	}
	public char peep() {
		if(!isEmpty())
			return items[top];
		return ' ';
	}
	private void resize() {
		char[] tmp = new char[items.length * 2];
		for(int i = 0; i < items.length; i++)
			tmp[i] = items[i];
		items = tmp;
		tmp = null;
	}
	
}
